test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -185,18 +192,34 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 			if err := externalMachine.Delete(ctx); err != nil {
 				log.Info("Failed to cleanup machine")
 			}
+			dockerMachine.Status.LoadBalancerConfigured = false
+			dockerMachine.Status.ImagePreLoaded = false
+			conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, infrav1.ContainerProvisioningFailedReason, clusterv1.ConditionSeverityWarning, "Re-provisioning")
+			conditions.Delete(dockerMachine, infrav1.BootstrapExecSucceededCondition)
+
 			res = ctrl.Result{RequeueAfter: 10 * time.Second}
 			retErr = nil
 		}
 	}()
 
-	if err := externalMachine.Create(ctx, role, machine.Spec.Version, dockerMachine.Spec.ExtraMounts); err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to create worker DockerMachine")
+	// Create the machine if not existing yet
+	if !externalMachine.Exists() {
+		if err := externalMachine.Create(ctx, role, machine.Spec.Version, dockerMachine.Spec.ExtraMounts); err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "failed to create worker DockerMachine")
+		}
+	}
+
+	// Update the ContainerProvisionedCondition condition
+	if externalMachine.Exists() {
```

Comment:
Nb. I'm implementing conditions.MarkTrue in a separate code branch vs the `if !externalMachine.Exists() {` branch above because I want to make the code re-entrant (vs settin condition true only after externalMachine.Create succeeded)

Reply:
It feels like this check is unnecessary though now, right? Like isn't it unreachable unless `externalMachine.Exists()`?